src/ray/core_worker/transport/raylet_transport.h

```diff
@@ -24,19 +25,43 @@ class CoreWorkerRayletTaskSubmitter : public CoreWorkerTaskSubmitter {
 
  private:
   /// Raylet client.
-  RayletClient &raylet_client_;
+  std::unique_ptr<RayletClient> &raylet_client_;
 };
 
-class CoreWorkerRayletTaskReceiver : public CoreWorkerTaskReceiver {
+class CoreWorkerRayletTaskReceiver : public CoreWorkerTaskReceiver,
+                                     public rpc::WorkerTaskHandler {
  public:
-  CoreWorkerRayletTaskReceiver(RayletClient &raylet_client);
+  CoreWorkerRayletTaskReceiver(std::unique_ptr<RayletClient> &raylet_client,
+                               boost::asio::io_service &io_service,
+                               rpc::GrpcServer &server);
 
   // Get tasks for execution from raylet.
   virtual Status GetTasks(std::vector<TaskSpec> *tasks) override;
 
+  /// TODO(zhijunfu): This is currently unused. Later when we migrate from worker "get
+  /// task" to raylet "assign task", this method will be used and the `GetTask` above will
+  /// be removed.
+  ///
+  /// Handle a `AssignTask` request.
+  /// The implementation can handle this request asynchronously. When hanling is done, the
+  /// `done_callback` should be called.
+  ///
+  /// \param[in] request The request message.
+  /// \param[out] reply The reply message.
+  /// \param[in] done_callback The callback to be called when the request is done.
```

Comment:
It seems brittle to require that `done_callback` be called inside `HandleAssignTask`. Can we instead pass in a lambda that calls `HandleAssignTask`, then the `done_callback`?

Reply:
I'm thinking about refactor `ServerCall` to call `done_callback` there instead, but as it would require changing node manager as well as object manager, I'd prefer to do it via a later PR.

Reply:
That doesn't seem necessary. I think we can just wrap `HandleAssignTask` in a lambda.

Reply:
sorry I'm not quite sure what the suggested change would be. May you add an example code? Thanks.

Reply:
Sure, I was suggesting that instead of passing in `&WorkerTaskHandler::HandleAssignTask` into `ServerCallFactoryImpl`, we could pass in something like:
```
[HandleAssignTask]( ... , done_callback) { HandleAssignTask(...); done_callback(); }
```

But actually, I think if I'm understanding the code correctly, what you're suggesting about doing it in `ServerCall` is the same thing. So agreed, let's just do it in a future PR.